### PR TITLE
Move compiled WW3 code to the case build directory 

### PR DESCRIPTION
**Pull request by erinethomas**
WW3 code currently compiles in the source code repo. This is severely problematic and creates several issues including: 
        a) cluttering the source code directory 
        b) creating problematic 'race-conditions' when more than one case with WW3 need to be compiled (such as running test suites)

This PR simply copies the required directories from the WW3 source code repo (the _aux_, _ftn_, and _bin_ directories) to the **"build/wav"** directory and compiles all WW3 code there. 
The compiled .F90 and executable tar files are now located here:  **"/build/wav/tmp/"** 

The E3SM commands _./case.build_  and _./case.build --clean-all_  correctly compile and clean all WW3 code.

### CODE REVIEW

**Review by jgfouca — APPROVED** [review]
Looks good. Was that warning about ww3 needing a lot of infrastructure updates wrong?

File: components/ww3/cime_config/buildlib_cmake [hunk 1]
```diff
@@ -43,14 +43,23 @@ def buildlib(bldroot, installpath, case):
     # Define WW3 repository directories
     repodir  = "{}/components/ww3/src".format(srcroot)
     modeldir = "{}/WW3/model".format(repodir)
+    builddir = "{}/wav".format(exeroot)
```
Comment (rljacob) [inline components/ww3/cime_config/buildlib_cmake:46]: Is this the CIME EXEROOT?

**Comment by erinethomas** [comment]
@jgfouca - I' not sure it was wrong exactly - but this PR provides an easy work around by copying the parts of ww3 needed to compile the code and then changing the way WW3 finds the new code... If we wanted to avoid copying the aux, bin, and ftn directories to the build directory I think we WOULD need to do a significant overhaul of the WW3 code ...

File: components/ww3/cime_config/buildlib_cmake (hunk 1 above)
Comment (erinethomas) [reply components/ww3/cime_config/buildlib_cmake:46]: exeroot here refers to the build directory .. i.e.  in env_build.xml:    <entry id="EXEROOT" value="/lcrc/group/e3sm/ac.ethomas/E3SMv3/Build_ww3_test.G.v3.PiControl/build">

Comment (erinethomas) [reply components/ww3/cime_config/buildlib_cmake:46]: @rljacob to clarify: this PR would then place the compiled wave code in the  "..../build/wav" directory.... if this is an inappropriate place to put the compiled WW3 code - suggestions for a better place are very welcome.

Comment (rljacob) [reply components/ww3/cime_config/buildlib_cmake:46]: Thats a good place!

File: components/ww3/cime_config/buildlib_cmake [hunk 2]
```diff
@@ -43,14 +43,23 @@ def buildlib(bldroot, installpath, case):
...
     # conditions if we were to try to build multiple ww3 cases simultaneously.
-    bindir   = "{}/bin".format(modeldir)
-    exedir   = "{}/exe".format(modeldir)
-    tmpdir   = "{}/tmp".format(modeldir)
+    bindir1   = "{}/bin".format(modeldir)
+    bindir   = "{}/wav/bin".format(exeroot)
+    shutil.copytree(bindir1, bindir)
+    auxdir1   = "{}/aux".format(modeldir)
+    auxdir   = "{}/wav/aux".format(exeroot)
+    shutil.copytree(auxdir1, auxdir)
```
Comment (sbrus89) [inline components/ww3/cime_config/buildlib_cmake:57]:
```suggestion
    auxdir_from   = "{}/aux".format(modeldir)
    auxdir_to   = "{}/aux".format(builddir)
    shutil.copytree(auxdir_from, auxdir_to)
```

File: components/ww3/cime_config/buildlib_cmake [hunk 3]
```diff
@@ -43,14 +43,23 @@ def buildlib(bldroot, installpath, case):
...
+    ftndir1   = "{}/ftn".format(modeldir)
+    ftndir   = "{}/wav/ftn".format(exeroot)
+    shutil.copytree(ftndir1, ftndir)
```
Comment (sbrus89) [inline components/ww3/cime_config/buildlib_cmake:60]:
```suggestion
    ftndir_from   = "{}/ftn".format(modeldir)
    ftndir_to   = "{}/ftn".format(builddir)
    shutil.copytree(ftndir_from, ftndir_to)
```

**Comment by erinethomas** [comment]
@sbrus89 Good suggestions - changes made to improve clarity.

**Review by sbrus89 — APPROVED** [review]
@erinethomas, I merged to next locally on Chrysalis and tested with `e3sm_wav_developer` defined as:

```
    "e3sm_wav_developer" : {
        "time"    : "0:45:00",
        "tests"   : (  
            "ERS.TL319_EC30to60E2r2_wQU225EC30to60E2r2.GMPAS-JRA1p5-WW3.ww3-jra_1958",
            "PEM.TL319_EC30to60E2r2_wQU225EC30to60E2r2.GMPAS-JRA1p5-WW3.ww3-jra_1958",
            "SMS_D_Ln3.TL319_EC30to60E2r2_wQU225EC30to60E2r2.GMPAS-JRA1p5-WW3.ww3-jra_1958",
            )    
    },   
```
Those tests all PASS when running `./create_test` in parallel.

**Comment by erinethomas** [comment]
closing to rename branch according to standards.

**Comment by rljacob** [comment]
Moved to #6663

**Pull request closed without merging**